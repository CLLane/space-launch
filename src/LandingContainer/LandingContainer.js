import React from "react";
import { Cards } from "../Cards/Cards";
import { connect } from "react-redux";

export const LandingContainer = ({ nextLaunch }) => {
  let landing = nextLaunch.map(launch => <Cards launch={launch} />);

  return <div className="landing-container">{landing}</div>;
};

export const mapStateToProps = ({ nextLaunch }) => ({
  nextLaunch
});

export default connect(mapStateToProps)(LandingContainer);
