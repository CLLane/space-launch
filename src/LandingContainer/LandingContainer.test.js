import React from "react";
import { shallow } from "enzyme";
import { LandingContainer, mapStateToProps } from "./LandingContainer";

describe('LandingContainer', () => {
  let wrapper;
  let mockNextLaunch = {
    id: 1,
    name: 'Chris'
  }
  
  beforeEach(() => {
    wrapper = shallow(<LandingContainer 
      nextLaunch={[mockNextLaunch]}
      key={1}
    />);
  });
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an object with a nextLuanch object', () => {
    const mockState = {
      nextLaunch: [{name: 'chris'}]
    }
    const expected = { 
      nextLaunch: [{ name: 'chris' }]
    }
    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })
})