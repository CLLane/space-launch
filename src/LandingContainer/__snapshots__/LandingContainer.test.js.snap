// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingContainer should match a snapshot 1`] = `
<div
  className="landing-container"
>
  <Cards
    key="Chris"
    launch={
      Object {
        "id": 1,
        "name": "Chris",
      }
    }
  />
</div>
`;
